### DISCUSSION TITLE
Echo cancellation features

### DISCUSSION THREAD

**Discussion by CollinBorns** [speechly/speechly#109] [Q&A] [answered]
How can I achieve echo cancellation in my application?

**Comment by aukkonen** [comment, marked as answer]
Speechly does not provide a separate mechanism for enabling or disabling echo cancellation. If you encounter echo or other acoustic issues, we recommend you to
1. Ensure that the microphone is properly positioned as close to the speaker as possible.
1. Ensure that the microphone settings (gain, noise cancellation, etc) are properly configured.
1. When using a Bluetooth headset, especially an older one, try to ensure that it uses a high-quality codec to transmit audio from the headset to the computer or mobile device. In particular, some older Bluetooth headsets will only transmit audio at 8kHz from the headset which may degrade speech recognition performance with Speechly.
